authz/providers/azure/rbac/rbac.go

```diff
@@ -443,33 +449,50 @@ func (a *AccessInfo) CheckAccess(request *authzv1.SubjectAccessReviewSpec) (*aut
 	}
 
 	// Fallback to managed namespace check
-	if !a.useManagedNamespaceResourceScopeFormat || a.clusterType != managedClusters {
-		klog.V(7).Infof("Checkaccess request is denied for user %s", checkAccessUsername)
-		return status, nil
-	}
+	managedNamespaceExists, managedNamespacePath := getManagedNameSpaceScope(request)
+	if a.useManagedNamespaceResourceScopeFormat &&
+		a.clusterType == managedClusters &&
+		managedNamespaceExists {
+		klog.V(7).Infof("Falling back to checking managed namespace scope for user %s", checkAccessUsername)
+		// Build managed namespace URL
+		managedNamespaceURL, err := buildCheckAccessURL(*a.apiURL, a.azureResourceId, true, managedNamespacePath)
+		if err != nil {
+			return nil, errors.Wrap(err, "error in building managed namespace check access URL")
+		}
 
-	klog.V(7).Infof("Falling back to checking managed namespace scope for user %s", checkAccessUsername)
-	exists, managedNamespacePath := getManagedNameSpaceScope(request)
-	if !exists {
-		klog.V(7).Infof(
-			"Skipping managed namespace check for user %s because subject access review is cluster scoped",
-			checkAccessUsername,
-		)
-		return status, nil
-	}
+		// Update resource IDs for managed namespace
+		for _, b := range checkAccessBodies {
+			b.Resource.Id = path.Join(a.azureResourceId, managedNamespacePath)
+		}
 
-	// Build managed namespace URL
-	managedNamespaceURL, err := buildCheckAccessURL(*a.apiURL, a.azureResourceId, true, managedNamespacePath)
-	if err != nil {
-		return nil, errors.Wrap(err, "error in building managed namespace check access URL")
+		status, err = a.performCheckAccess(managedNamespaceURL, checkAccessBodies, checkAccessUsername)
+		if err != nil {
+			return nil, err
+		}
+		if status != nil && status.Allowed {
+			klog.V(7).Infof("Managed namespace checkaccess request is allowed for user %s", checkAccessUsername)
+			return status, nil
+		}
 	}
 
-	// Update resource IDs for managed namespace
-	for _, b := range checkAccessBodies {
-		b.Resource.Id = path.Join(a.azureResourceId, managedNamespacePath)
+	// Fallback to fleet scope check when the managedCluster has joined a fleet
+	if found, fleetMgrID := a.getFleetManagerResourceID(); found {
+		fleetURL, err := buildCheckAccessURL(*a.apiURL, fleetMgrID, managedNamespaceExists, managedNamespacePath)
```

Comment:
if the mc is not member of the fleet, even there is a role assignment for a fleet managed namespace (e.g. /subscriptions/12345678-1234-1234-1234-123456789abc/resourceGroups/testResourceGroup/providers/Microsoft.ContainerService/fleets/my-fleet/managedNamespaces/{ns1}, will it get access?